lib/OpenQA/Git.pm

```diff
@@ -21,45 +22,68 @@ sub config ($self, $args = undef) {
     return $app->config->{'scm git'};
 }
 
-sub _prepare_git_command ($self, $args = undef) {
-    my $dir = $args->{dir} // $self->dir;
-    if ($dir !~ /^\//) {
-        my $absolute_path = abs_path($dir);
-        $dir = $absolute_path if ($absolute_path);
+sub _validate_attributes ($self) {
+    for my $mandatory_property (qw(app dir user)) {
+        die "no $mandatory_property specified" unless $self->$mandatory_property();
     }
-    return ('git', '-C', $dir);
 }
 
-sub _format_git_error ($result, $error_message) {
-    if ($result->{stderr} or $result->{stdout}) {
-        $error_message .= ': ' . $result->{stdout} . $result->{stderr};
+sub _run_cmd ($self, $args, $options = {}) {
+    my $expand_git_path = $options->{expand_git_path} // 1;
+    my $use_ssh = $options->{use_ssh} // 0;
```

Comment:
I think `use_ssh` is the wrong word. It's not abuot using ssh, but to tell shh to use BatchMode. But that should be done in a followup, so that we don't block other tickets.

Reply:
Changed to `ssh_batchmode`, is this ok for now ? Now sure how this change would block other tickets ?

Reply:
Yeah.
I just meant the longer we wait with this PR, the longer Dominik has to wait with his PR touching the same code.